File: apps/party-time-backend/src/main/java/com/partytime/configuration/security/PartyTimeUserDetails.java
package com.partytime.configuration.security;

import com.partytime.jpa.entity.Account;
import org.springframework.security.core.userdetails.User;

import java.util.Collections;

public class PartyTimeUserDetails extends User {

    public PartyTimeUserDetails(Account account) {
        super(account.getEmail(), account.getPwHash(), Collections.emptyList());
    }

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/configuration/security/PartyTimeUserDetails.kt
package com.partytime.configuration.security

import com.partytime.jpa.entity.Account
import org.springframework.security.core.userdetails.User

/**
 * User details used to authenticate users when they want to use protected server functionality.
 * Server-internal use only.
 */
class PartyTimeUserDetails(account: Account): User(
    account.email, account.pwHash, emptyList()
)
